calliope/analysis/plotting/energy_flows.py

```diff
@@ -0,0 +1,319 @@
+    tech_names = set(model._model_data.techs.values)
+
+    production_data = []
+    # we iterate through each dimension for one timestep in order to
+    # add the different production sources of the location and line
+    # transmissions toward it
+    # Complexity: O(len(carriers)*len(techs)*len(locations)). Considering
+    # len(carriers) is supposed to be small (<10), we have a large margin in
+    # terms of techs and locations numbers (len(techs)*len(locations) <= 10^9,
+    # equivalent to one second processing time for python)
+    for location in locs_techs_carriers_production.locs.values:
+        for carrier in locs_techs_carriers_production.carriers.values:
+            techs_production = locs_techs_carriers_production.sel(
+                carriers=carrier,
+                locs=location
+            ).to_pandas()
+            for tech, prod in techs_production.loc[:, timestep].iteritems():
+                if prod and prod > 0:
+                    # only add legend information once for a tech
+                    tech_name = tech.split(":")[0]
+                    if tech_name in tech_names:
+                        add_legend = True
+                        tech_names.discard(tech_name)
+                        name = model._model_data.names.loc[tech_name].item()
+                    else:
+                        add_legend = False
+                        name = ''
+                    # if some energy is at stake
+                    if len(tech.split(":")) > 1:
+                        # "transmission_type:location"
+                        # if it gets energy from another location
+                        production_data.extend(
+                            _line(
+                                loc_coordinates, location,
+                                carrier, tech, prod,
+                                scale_factor, techs_colors,
+                                timestep == timesteps[0],
+                                add_legend, name
+                            )
+                        )
+                    else:
+                        # if the energy comes from this location
+                        production_data.extend(
+                            _marker(
+                                loc_coordinates, location,
+                                carrier, tech, prod,
+                                scale_factor, techs_colors,
+                                timestep == timesteps[0],
+                                add_legend, name
+                            )
+                        )
+
+    return production_data
+
+
+def plot_energy_flows(model, timestep_cycle=1,
+                      timestep_index_subset=[], **kwargs):
+    """
+    Parameters
+    ----------
+    timestep_cycle : int, optional
+        Shows one of every timestep_cycle timesteps. Default is 1 (all timesteps
+        are shown).
+    timestep_index_subset : list of int, optional
```

Comment:
Could this subsetting be done in a way more similar to the other plotting functions? If not with a generic ``subset`` param (see snippet below from ``plot_timeseries``) at least with the ability to pass a datetime range or a start and end datetime like ``['2018-01-01', '2018-01-31']``?

```
    subset : dict, optional
        Dictionary by which data is subset (uses xarray `loc` indexing). Keys
        any of ['timeseries', 'locs', 'techs', 'carriers', 'costs'].
```

Reply:
Yes it should be done like this, I agree